crates/driver/src/tests/setup/mod.rs

```diff
@@ -679,7 +679,12 @@ impl Test {
         let status = res.status();
         let body = res.text().await.unwrap();
         tracing::debug!(?status, ?body, "got a response from /solve");
-        Solve { status, body }
+        Solve {
+            status,
+            body,
+            fulfillments: &self.fulfillments,
+            blockchain: &self.blockchain,
```

Comment:
I think these fields should not be populated by the values used to setup/configure the test. Otherwise the solver might return something completely different and we'd not see it when comparing our expectation against it.
Looks like the existing code currently solves this by having sth like `.fulfillments()` on the `SolveOk` struct that parses `body` just in time.

Reply:
Can you elaborate on what you mean by this? Before those fields were populated on the `reveal()` method below (because we only checked the expected orders there) and I cannot see how the approach changed nor how a jit `.fulfillments()` on `SolveOk` would work.